src/sentry/api/paginator.py

```diff
@@ -354,9 +354,15 @@ def get_result(self, limit, cursor=None):
         offset = cursor.offset if cursor is not None else 0
         # Request 1 more than limit so we can tell if there is another page
         data = self.data_fn(offset=offset, limit=limit + 1)
-        has_more = (len(data) == limit + 1)
-        if has_more:
-            data.pop()
+
+        if isinstance(data, list):
+            has_more = len(data) == limit + 1
+            if has_more:
+                data.pop()
+        else:
```

Comment:
I'd probably do `elif isinstance(data.get('data'), list)` here, and add a comment / update the docstring above explaining why we need to do this